appveyor.yml

```diff
@@ -1,11 +1,65 @@
 image: Visual Studio 2015
 
+environment:
+  NUGET_API_KEY:
+    secure: Rh/UHYn7HGj8a7LFWo9C1rSA0sHSQNV3wIqj9bvwRR8yLmCXu9JmbNkmkctk2cBn
+  MYGET_API_KEY:
+    secure: hA4Ut1N2lrrdEtAN24Bty/FNiU0d/Ur/dLYSqpr8jKHOvoO7MU4jD+KwzUvATh+E
+
 pull_requests:
   do_not_increment_build_number: true
 
 build_script:
 - ps: |
-    & .\build.cmd CompleteBuild
+    # A trick to test if variable is defined. Number could never be null.
+    $isPullRequest = -not ($Env:APPVEYOR_PULL_REQUEST_NUMBER -eq $null)
+    $isTagBuild = $Env:APPVEYOR_REPO_TAG -eq "true"
+    $branchName = $Env:APPVEYOR_REPO_BRANCH
+
+    Write-Host "[BUILD CONTEXT INFO] Is PR: $isPullRequest, is tag: $isTagBuild, branch name: '$branchName'" -ForegroundColor Green
+
+    if($isTagBuild)
+    {
+      $tagName = $Env:APPVEYOR_REPO_TAG_NAME
+      $isSemVerTag = $tagName -match '^v.*'
+      # We consider version as alpha if it contains the "-alpha" pre-release tag.
+      $isAlphaVerTag = $tagName -match '^v[\d\.]+-alpha[\d\.]*$'
```

Comment:
I could redesign it and alter the version generation logic to use branch name as a pre-release tag, but only if we have a really good reason for that.

[Previously](https://github.com/AutoFixture/AutoFixture/pull/753) we agreed that we will push the `vNext` branch to the private feed only. I don't think that single `alpha` tag for each `vNext` is actually too much (e.g. `v4.0.0-alpha`, `v5.0.0-alpha`, etc). Even if we have 5 more major versions which will take a couple of years, we will have 5 more tags. Given that we generate a tag per each release, you will not even notice these tags among 164 tags (!!) we currently have for `v3`.

Also I would like to keep version generation logic as simple as possible, so it's easily to understand what is going on. If I start to mix branch name, it will complicate the logic too much.

Do you still see that single `alpha` tag for each `vNext` is too much?